lib/compilers/solidity.ts

```diff
@@ -115,20 +115,18 @@ export class SolidityCompiler extends BaseCompiler {
                                 return node.name === 'FunctionDefinition';
                             })
                             .map(node => {
-                                const [begin, length] = node.src.split(':').map(x => parseInt(x));
+                                const [begin, length] = node.src.split(':').map(x => Number.parseInt(x));
 
                                 let name = node.attributes.isConstructor ? 'constructor' : node.attributes.name;
 
                                 // encode the args into the name so we can
                                 // differentiate between overloads
                                 if (node.children[0].children.length > 0) {
-                                    name +=
-                                        '_' +
-                                        node.children[0].children
-                                            .map(paramNode => {
-                                                return paramNode.attributes.type;
-                                            })
-                                            .join('_');
+                                    name += `_${node.children[0].children
```

Comment:
This is beyond ridiculous, can we limit the amount of code crammed into the backquotes?

Reply:
Will take care of this with the other template stuff